include/slang/ast/expressions/AssertionExpr.h

```diff
@@ -190,6 +213,22 @@ struct SequenceRange {
                                     const ASTContext& context, bool allowUnbounded);
 
     void serializeTo(ASTSerializer& serializer) const;
+
+    bool operator!=(const SequenceRange& right) const;
+
+    bool operator==(const SequenceRange& right) const;
+
+    bool operator>(const SequenceRange& right) const;
```

Comment:
You can instead implement all of the comparison operators by implementing the "spaceship" operator: `operator<=>` to simplify this a bunch.

Reply:
currently it's failed on MacOS build - I'm working with it

Reply:
I chose to implement only the `operator<` and changed the comparison logic

Reply:
Spaceship (`<=>`) operator requires realization for any field but for `std::optional` that operator marked as deleted. It requires for changing processing of `SequenceRange`s with unbonded maximum I don't have any good idea for resolving with another way.